### PR TITLE
VCP & IAS

### PR DESCRIPTION
**Pull request by szymon-czapracki**
Add tests for VCP and IAS.

### CODE REVIEW

File: autopts/wid/ias.py
Comment (michal-narajowski) [inline autopts/wid/ias.py:56]:
```suggestion
    return stack.ias.alert_lvl == 2
```

File: autopts/wid/ias.py
Comment (michal-narajowski) [inline autopts/wid/ias.py:66]:
```suggestion
    return stack.ias.alert_lvl == 0
```

File: autopts/wid/ias.py
Comment (michal-narajowski) [inline autopts/wid/ias.py:77]:
```suggestion
    return stack.ias.alert_lvl == 1
```

File: autopts/wid/ias.py
Comment (michal-narajowski) [inline autopts/wid/ias.py:88]:
```suggestion
    stack.ias.alert_lvl == 1
```

**Review by michal-narajowski — CHANGES_REQUESTED** [review]
A few comments, but otherwise LGTM

File: autopts/ptsprojects/zephyr/aics.py
Comment (szymon-czapracki) [reply autopts/ptsprojects/zephyr/aics.py]: Done

File: autopts/ptsprojects/zephyr/ias.py
Comment (szymon-czapracki) [reply autopts/ptsprojects/zephyr/ias.py]: Done

File: autopts/ptsprojects/zephyr/ias.py
Comment (szymon-czapracki) [reply autopts/ptsprojects/zephyr/ias.py:86]: Done

File: autopts/ptsprojects/zephyr/aics.py
Comment (szymon-czapracki) [reply autopts/ptsprojects/zephyr/aics.py:87]: Done

File: autopts/ptsprojects/zephyr/vcs.py
Comment (szymon-czapracki) [reply autopts/ptsprojects/zephyr/vcs.py]: Done

File: autopts/ptsprojects/zephyr/vocs.py
Comment (szymon-czapracki) [reply autopts/ptsprojects/zephyr/vocs.py]: Done

File: autopts/pybtp/btp/gap.py
Comment (szymon-czapracki) [reply autopts/pybtp/btp/gap.py]: Silly mistake, sorry - removed.

File: autopts/pybtp/defs.py
Comment (szymon-czapracki) [reply autopts/pybtp/defs.py]: Done

File: autopts/wid/ias.py
Comment (szymon-czapracki) [reply autopts/wid/ias.py]: Done

File: autopts/wid/ias.py
Comment (szymon-czapracki) [reply autopts/wid/ias.py]: Done

File: autopts/wid/ias.py
Comment (szymon-czapracki) [reply autopts/wid/ias.py]: Done

File: autopts/wid/ias.py
Comment (szymon-czapracki) [reply autopts/wid/ias.py]: Done

File: autopts/ptsprojects/stack.py
Comment (sjanc) [inline autopts/ptsprojects/stack.py:1150]: those are bitfields (bit number match service ID)

        Opcode 0x02 - Read Supported Services command/response

                Controller Index:       <non-controller>
                Command parameters:     <none>
                Response parameters:    <supported services> (variable)

                Each bit in response is a flag indicating if service with ID
                matching bit number is supported. Bit set to 1 means that
                service is supported. If specific bit is not present in response
                (less than required bytes received) it shall be assumed that
                service is not supported.

File: autopts/pybtp/defs.py
Comment (sjanc) [inline autopts/pybtp/defs.py:27]: this is conflicting with gatt server service

**Review by sjanc — CHANGES_REQUESTED** [review]
BTP documentation is missing

File: autopts/ptsprojects/zephyr/vocs.py
Comment (MariuszSkamra) [inline autopts/ptsprojects/zephyr/vocs.py:79]: Shouldn't you register VOCS instead?

File: autopts/ptsprojects/zephyr/aics.py
Comment (MariuszSkamra) [inline autopts/ptsprojects/zephyr/aics.py:80]: Shouldn't you register AICS instead? The tests are testing AICS functionality, not VCP, thus you should expose an API that is for AICS imo as the test cases are testing AICS layer only. You could still have a implementation that has AICS without VCS. BTP is meant to be testing protocol used across all platforms.

**Review by MariuszSkamra — CHANGES_REQUESTED** [review]

**Review by michal-narajowski — CHANGES_REQUESTED** [review]
A few more comments. I didn't notice them the first time around.

File: autopts/ptsprojects/stack.py
Comment (szymon-czapracki) [reply autopts/ptsprojects/stack.py]: done

File: autopts/pybtp/defs.py
Comment (szymon-czapracki) [reply autopts/pybtp/defs.py]: done

File: autopts/ptsprojects/zephyr/vocs.py
Comment (szymon-czapracki) [reply autopts/ptsprojects/zephyr/vocs.py]: done

File: autopts/ptsprojects/zephyr/aics.py
Comment (szymon-czapracki) [reply autopts/ptsprojects/zephyr/aics.py]: done

File: autopts/ptsprojects/stack.py
Comment (szymon-czapracki) [reply autopts/ptsprojects/stack.py]: done

File: autopts/ptsprojects/stack.py
Comment (szymon-czapracki) [reply autopts/ptsprojects/stack.py]: done

File: autopts/ptsprojects/stack.py
Comment (szymon-czapracki) [reply autopts/ptsprojects/stack.py]: done

**Review by michal-narajowski — APPROVED** [review]

File: doc/btp_vcs.txt
Comment (sjanc) [inline doc/btp_vcs.txt:22]: same here, do we need explicit init?

File: doc/btp_aics.txt
Comment (sjanc) [inline doc/btp_aics.txt:32]: "Response parameters:    <none>"     (same for other commands that return only status)

File: doc/btp_aics.txt
Comment (sjanc) [inline doc/btp_aics.txt:86]: 0x0a  :-)

File: doc/btp_aics.txt
Comment (szymon-czapracki) [reply doc/btp_aics.txt]: We don't - removed.

File: doc/btp_aics.txt
Comment (szymon-czapracki) [reply doc/btp_aics.txt:24]: Done

File: doc/btp_aics.txt
Comment (szymon-czapracki) [reply doc/btp_aics.txt]: Done (sic!)

File: autopts/ptsprojects/stack.py
Comment (MariuszSkamra) [inline autopts/ptsprojects/stack.py:1039]:
```suggestion
class IAS:
    ALERT_LEVEL_NONE = 0
    ALERT_LEVEL_MILD = 1
    ALERT_LEVEL_HIGH = 2

    def __init__(self):
        self.alert_lvl = self.ALERT_LEVEL_NONE

    def is_mild_alert_set(self, args):
        return self.alert_lvl == self.ALERT_LEVEL_MILD

    def is_high_alert_set(self, args):
        return self.alert_lvl == self.ALERT_LEVEL_HIGH

    def is_alert_stopped(self, args):
        return self.alert_lvl == self.ALERT_LEVEL_NONE
```

File: autopts/ptsprojects/stack.py
Comment (MariuszSkamra) [inline autopts/ptsprojects/stack.py:1205]:
```suggestion
        if self.vcs:
            self.vcs_init()

        if self.aics:
            self.aics_init()

        if self.vocs:
            self.vocs_init()

        if self.ias:
            self.ias_init()

```

File: autopts/ptsprojects/zephyr/aics.py
Comment (MariuszSkamra) [inline autopts/ptsprojects/zephyr/aics.py:81]:
```suggestion
                      TestFunc(stack.aics_init)]
```

File: autopts/ptsprojects/zephyr/ias.py
Comment (MariuszSkamra) [inline autopts/ptsprojects/zephyr/ias.py:80]:
```suggestion
                      TestFunc(stack.ias_init)]
```

File: autopts/ptsprojects/zephyr/vcs.py
Comment (MariuszSkamra) [inline autopts/ptsprojects/zephyr/vcs.py:81]:
```suggestion
                      TestFunc(stack.vcs_init)]
```

File: autopts/ptsprojects/zephyr/vocs.py
Comment (MariuszSkamra) [inline autopts/ptsprojects/zephyr/vocs.py:80]:
```suggestion
                      TestFunc(stack.vocs_init)]
```

File: autopts/pybtp/btp/btp.py
Comment (MariuszSkamra) [inline autopts/pybtp/btp/btp.py:626]: What about handling events from other services you add (VCS, VOCS, AICS)?

File: autopts/pybtp/btp/vcs.py
Comment (MariuszSkamra) [inline autopts/pybtp/btp/vcs.py:62]: Why in some cases you wait for response while in other not? 
The BTP is defines those two operations the same way:
```

	Opcode 0x05 - Mute
		Controller Index:	<controller id>
		Command parameter:	<none>
		Response parameters:	<none>

		This command is used for setting mute value to "mute" in VCS.

		In case of an error, the error response will be returned.

	Opcode 0x06 - Unmute
		Controller Index:	<controller id>
		Command parameter:	<none>
		Response parameters:	<none>

		This command is used for setting mute value to "unmute" in VCS.

		In case of an error, the error response will be returned.
```

**Review by MariuszSkamra — CHANGES_REQUESTED** [review]

File: autopts/pybtp/btp/vcs.py
Comment (szymon-czapracki) [reply autopts/pybtp/btp/vcs.py]: Removed - we don't wait for the response now.

File: autopts/pybtp/btp/btp.py
Comment (szymon-czapracki) [reply autopts/pybtp/btp/btp.py:630]: Done

File: autopts/ptsprojects/zephyr/vocs.py
Comment (szymon-czapracki) [reply autopts/ptsprojects/zephyr/vocs.py]: Done

File: autopts/ptsprojects/zephyr/vcs.py
Comment (szymon-czapracki) [reply autopts/ptsprojects/zephyr/vcs.py]: Done

File: autopts/ptsprojects/zephyr/ias.py
Comment (szymon-czapracki) [reply autopts/ptsprojects/zephyr/ias.py]: Done

File: autopts/ptsprojects/stack.py
Comment (szymon-czapracki) [reply autopts/ptsprojects/stack.py]: Done

File: autopts/ptsprojects/stack.py
Comment (szymon-czapracki) [reply autopts/ptsprojects/stack.py]: Done

File: autopts/pybtp/btp/vcs.py
Comment (sjanc) [reply autopts/pybtp/btp/vcs.py]: hmm why?:) this is like command status, it should come immediately and if BTP failed we should handle it...

**Comment by michal-narajowski** [comment]
Is this ready for another review? Can you fix the failing check?

**Comment by szymon-czapracki** [comment]
> Is this ready for another review? Can you fix the failing check?

Ready for another round.

**Comment by codecoup-tester** [comment]
Scheduled PR #https://github.com/intel/auto-pts/pull/840#issuecomment-1412355071 after 17:34:51.

**Comment by codecoup-tester** [comment]
AutoPTS Bot results:
Failed tests:
AICS    AICS/SR/SGGIT/CP/BI-03-C        BTP ERROR
AICS    AICS/SR/SGGIT/CP/BI-04-C        BTP ERROR
AICS    AICS/SR/SPN/BV-01-C             BTP ERROR
AICS    AICS/SR/SPN/BV-02-C             BTP ERROR
VCS     VCS/SR/CP/BV-01-C               INCONC
VCS     VCS/SR/CP/BV-02-C               INCONC
VCS     VCS/SR/CP/BV-03-C               INCONC
VCS     VCS/SR/CP/BV-04-C               INCONC
VOCS    VOCS/SR/SPN/BV-02-C             FAIL

**Review by sjanc — APPROVED** [review]
We will probably end up with merging zephyr-leaudio in zephyr-master to keep single project and make lanuch studio dependency check easier.

But that can be done after this PR is merged.

**Review by MariuszSkamra — APPROVED** [review]

**Pull request merged**
